File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        dictionary = {}

        for string in strs:
            count_arr = [0] * 26

            for letter in string:
                count_arr[ord(letter) - ord('a')] += 1

            key = tuple(count_arr)

            if key not in dictionary:
                dictionary[key] = []

            dictionary[key].append(string)

        return dictionary.values()

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
      unordered_map<string,vector<string>> map;
      int n=strs.size();
      for(int i=0;i<n;i++){
        string temp=strs[i];
        sort(temp.begin(),temp.end());
        map[temp].push_back(strs[i]);

      }
      vector<vector<string>>ans;
      for(auto it:map){
        ans.push_back(it.second);
      }
      return  ans;
    }
};